src/main/java/picard/analysis/StrandSpecificityUtil.java

```diff
@@ -0,0 +1,196 @@
+            });
+
+            // Only examine reads that overlap an exon
+            if (overlapsExon) {
+                // We would need both read pairs to check multiple genes, and therefore we ignore such cases and call
+                // them unclassified later.
+                if (overlappingGenes.stream().map(Gene::isPositiveStrand).distinct().count() == 1) {
+                    final boolean negativeReadStrand            = rec.getReadNegativeStrandFlag();
+                    final boolean negativeTranscriptionStrand   = overlappingGenes.iterator().next().isNegativeStrand();
+
+                    if (!rec.getReadPairedFlag() || rec.getFirstOfPairFlag()) {
+                        if (negativeReadStrand == negativeTranscriptionStrand) readOneAgree    += 1;
+                        else                                                   readOneDisagree += 1;
+                    } else {
+                        if (negativeReadStrand == negativeTranscriptionStrand) readTwoAgree    += 1;
+                        else                                                   readTwoDisagree += 1;
+                    }
+                }
+                numReadsExamined += 1;
+            }
+        }
+
+        // NB: these two values may not add up to one if some reads cannot be classified due to overlapping multiple
+        // genes that have different transcription strands
+        double fractionFirstReadTranscriptionStrand  = (readOneAgree + readTwoDisagree) / (double) numReadsExamined;
+        double fractionSecondReadTranscriptionStrand = (readOneDisagree + readTwoAgree) / (double) numReadsExamined;
+        double fractionUnclassified = (numReadsExamined - readOneAgree - readOneDisagree - readTwoAgree - readTwoDisagree) / (double) numReadsExamined;
+
+        final int numReads = numReadsExamined; // OK IntelliJ
+        log.ifPresent(l -> {
+            l.info("Examined " + numReads + " reads to infer strand specificity.");
+            l.info("Fraction of reads consistent with read one on the transcription strand:     " + fractionFirstReadTranscriptionStrand);
+            l.info("Fraction of reads consistent with read one not on the transcription strand: " + fractionSecondReadTranscriptionStrand);
+            l.info("Unclassified reads: " + fractionUnclassified);
+        });
+
+        if (fractionFirstReadTranscriptionStrand >= minFraction && fractionSecondReadTranscriptionStrand >= minFraction) {
+            throw new IllegalStateException("Could not infer strand specificity as both + ("
+                    + fractionFirstReadTranscriptionStrand
+                    + ") and - ("
+                    + fractionSecondReadTranscriptionStrand
+                    + ") are >= the threshold '" + minFraction + "'");
+        }
+        else if (fractionUnclassified >= minFraction) {
+            throw new IllegalStateException("Could not infer strand specificity as the number of unclassified reads ("
```

Comment:
I don't think this can happen - and even if it does, should it really throw an exception and not just return `NONE`?